libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +82,193 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
   public buttonContainerElement: ElementRef<HTMLElement> | undefined;
 
   @ViewChild('defaultButton', { read: ElementRef })
-  public defaultButtonElement: ElementRef<HTMLElement> | undefined;
+  public defaultButtonElement: ElementRef<HTMLButtonElement> | undefined;
 
   @ContentChild(ButtonComponent, { read: ElementRef }) public userProvidedButton:
-    | ElementRef<HTMLElement>
+    | ElementRef<HTMLButtonElement>
     | undefined;
 
   @ViewChild(FloatingDirective)
-  private floatingDirective: FloatingDirective;
+  _floatingMenu: FloatingDirective;
 
-  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
+  @ContentChildren(ItemComponent, { read: ElementRef }) public kirbyItems: QueryList<
+    ElementRef<HTMLElement>
+  >;
 
+  public floatingMenuIsShown: boolean = false;
+  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
   private scrollListenerDisposeFn: EventListenerDisposeFn;
+  private focusedIndex = -1;
+
+  @HostListener('keydown', ['$event'])
+  _onKeydown(event: KeyboardEvent) {
+    if (this.floatingMenuIsShown) {
+      this.handleOnKeyForOpenedMenu(event);
+    } else {
+      this.handleOnKeyForClosedMenu(event);
+    }
+  }
+
+  private preventFurtherPropagation(event: KeyboardEvent) {
+    event.stopImmediatePropagation();
+    event.preventDefault();
+  }
+
+  private getFirstInteractiveElement(el: HTMLIonItemElement) {
+    const controls = el.querySelectorAll<
+      HTMLIonToggleElement | HTMLIonRadioElement | HTMLIonCheckboxElement
+    >('ion-toggle:not([disabled]), ion-checkbox:not([disabled]), ion-radio:not([disabled])');
+
+    return controls[0] || undefined;
```

Comment:
Index accessing an empty list already returns undefined: 
```suggestion
    return controls[0];
```